### PR TITLE
fixed error messages and inconsistent if-condition-raise statements

### PR DESCRIPTION
**Pull request by islem-esi**

### CODE REVIEW

File: tensorflow/python/debug/cli/command_parser.py [hunk 1]
```diff
@@ -268,10 +268,10 @@ def parse_ranges(range_string):
     if len(item) != 2:
       raise ValueError("Incorrect number of elements in range")
     elif not isinstance(item[0], (int, float)):
-      raise ValueError("Incorrect type in the 1st element of range: %s" %
+      raise TypeError("Incorrect type in the 1st element of range: %s" %
```
Comment (rohan100jain) [inline tensorflow/python/debug/cli/command_parser.py:271]: Lets keep these as ValueError? Or we'll have to update the Raises: section as well. Typically in TF we just throw ValueError for any input validation problems.

File: tensorflow/python/debug/cli/command_parser.py [hunk 2]
```diff
@@ -268,10 +268,10 @@ def parse_ranges(range_string):
...
                        type(item[0]))
     elif not isinstance(item[1], (int, float)):
-      raise ValueError("Incorrect type in the 2nd element of range: %s" %
+      raise TypeError("Incorrect type in the 2nd element of range: %s" %
```
Comment (rohan100jain) [inline tensorflow/python/debug/cli/command_parser.py:274]: ditto

File: tensorflow/python/debug/cli/command_parser_test.py [hunk 3]
```diff
@@ -300,11 +300,11 @@ def testParseInvalidRangeString(self):
                                 "Incorrect number of elements in range"):
       command_parser.parse_ranges("[inf]")
 
-    with self.assertRaisesRegex(ValueError,
+    with self.assertRaisesRegex(TypeError,
```
Comment (rohan100jain) [inline tensorflow/python/debug/cli/command_parser_test.py:303]: Lets keep it as ValueError

File: tensorflow/python/debug/cli/command_parser_test.py [hunk 4]
```diff
@@ -300,11 +300,11 @@ def testParseInvalidRangeString(self):
...
                                 "Incorrect type in the 1st element of range"):
       command_parser.parse_ranges("[1j, 1]")
 
-    with self.assertRaisesRegex(ValueError,
+    with self.assertRaisesRegex(TypeError,
```
Comment (rohan100jain) [inline tensorflow/python/debug/cli/command_parser_test.py:307]: same here

**Review by rohan100jain — CHANGES_REQUESTED** [review]
In general, lets keep it as ValueError but thanks for improving the error messages!

File: tensorflow/python/debug/cli/command_parser.py (hunk 1 above)
Comment (islem-esi) [reply tensorflow/python/debug/cli/command_parser.py:271]: Oh, I see, I was just following the Python documentation :

> ValueError
> 
> Raised when an operation or function receives an argument **that has the right type** but an **inappropriate value**, and the situation is not described by a more precise exception such as [IndexError](https://docs.python.org/3/library/exceptions.html#IndexError).

**Comment by gbaned** [comment]
@islem-esi Can you please resolve conflicts? Thank you!

**Comment by gbaned** [comment]
Hi @islem-esi Can you please check @rohan100jain's [comments](https://github.com/tensorflow/tensorflow/pull/58589#issuecomment-1372698939) and keep us posted ? Thank you!

**Pull request closed without merging**